rcl/src/rcl/timer.c

```diff
@@ -49,6 +49,8 @@ typedef struct rcl_timer_impl_t
   atomic_bool canceled;
   // The user supplied allocator.
   rcl_allocator_t allocator;
+  // A guard condition used to wake a wait set to reset cancelled timer.
+  rcl_guard_condition_t notify_guard_condition;
```

Comment:
I don't see a reason to use a different guard condition.
Guard conditions just wakes the wait set, it isn't used to check if the timer is ready or not.

Reply:
> I don't see a reason to use a different guard condition.
> Guard conditions just wakes the wait set, it isn't used to check if the timer is ready or not.

I removed notify_guard_condition and always add a guard condition of timer to node wait set.